src/main/java/com/hubspot/jinjava/interpret/JinjavaInterpreter.java

```diff
@@ -280,11 +296,15 @@ public String render(Node root) {
    *          node to render
    * @param processExtendRoots
    *          if true, also render all extend parents
+   * @param renderLimit
+   *          the number of characters the result may contain
    * @return rendered result
    */
-  public String render(Node root, boolean processExtendRoots) {
-    OutputList output = new OutputList(config.getMaxOutputSize());
-
+  private String render(Node root, boolean processExtendRoots, long renderLimit) {
+    long safeRenderSize = (config.getMaxOutputSize() == 0)
+      ? renderLimit
+      : Math.min(renderLimit, config.getMaxOutputSize());
+    OutputList output = new OutputList(safeRenderSize);
```

Comment:
Someone could pass a negative or 0-value render limit and bypass the max output size.

Reply:
And it would be good to test these different cases for ensuring render limit doesn't allow rendering more than configured

Reply:
So by default, this using 0 (unlimited) as the render size makes me wonder what the problem is here.  Do we have cases where we modify this config value for limited length rendering?  As per my logic here in the filter, if no value/an invalid value is provided, we just pass in 0 anyway, so that to me seems like an easy workaround.

Right now, `config.getMaxOutputSize()` returns 0.

Reply:
If `config.getMaxOutputSize()` is 100 and `renderLimit` is 0, then it will bypass the configured limit and render unlimited characters.
For HubL, we are always setting a non-zero value for `config.getMaxOutputSize()`